### PR TITLE
vis: tweaks

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

🙃 🔫.

### CODE REVIEW

File: Formula/vis.rb [hunk 1]
```diff
@@ -40,10 +40,16 @@ def install
 
     system "./configure", "--prefix=#{libexec}"
     system "make", "install"
+
+    # Install the tools in bin under their default names since
+    # there's no clash with the system there and they expect
+    # to be executed from within the $PATH.
+    bin.install Dir[libexec/"bin/*"].reject { |f| f =~ /vis$/ }
```
Comment (ilovezfs) [inline Formula/vis.rb]: why not set --prefix=#{prefix} and then only move vis to libexec?

Comment (DomT4) [reply Formula/vis.rb]: Would need to rename the manpage then as well. Wasn't sure how much renaming you were happy with. If you're alright with that happy to flip it.

Comment (DomT4) [reply Formula/vis.rb]: I've made the chance but not force pushed, so there's a chance for you to decide if you hate it or not.

File: Formula/vis.rb [hunk 2]
```diff
@@ -38,12 +38,16 @@ def install
       system "luarocks", "build", "lpeg", "--tree=#{luapath}"
     end
 
-    system "./configure", "--prefix=#{libexec}"
+    system "./configure", "--prefix=#{prefix}"
     system "make", "install"
+
+    # Rename vis & the matching manpage to avoid clashing with the system.
+    man1.install man1/"vis.1" => "vise.1"
```
Comment (ilovezfs) [inline Formula/vis.rb]: `mv` is probably a little nicer

File: Formula/vis.rb [hunk 3]
```diff
@@ -38,12 +38,16 @@ def install
...
+    (libexec/"bin").install bin/"vis"
```
Comment (ilovezfs) [inline Formula/vis.rb]: maybe use env_script_all_files if we're having to export anyway and then `mv` vis?

File: Formula/vis.rb [hunk 4]
```diff
@@ -38,12 +38,16 @@ def install
...
     (bin/"vise").write <<~EOS
       #!/bin/sh
-      LUA_PATH="#{ENV["LUA_PATH"]}"
-      LUA_CPATH="#{ENV["LUA_CPATH"]}"
+      export LUA_PATH="#{ENV["LUA_PATH"]}"
+      export LUA_CPATH="#{ENV["LUA_CPATH"]}"
       VIS_BASE=#{libexec}
```
Comment (ilovezfs) [inline Formula/vis.rb]: Is this still correct if we use prefix not libexec?

File: Formula/vis.rb (hunk 2 above)
Comment (DomT4) [reply Formula/vis.rb]: Aye. I can never make my mind up which looks less grim in these situations, so happy with `mv` if that's your preference.

File: Formula/vis.rb (hunk 4 above)
Comment (DomT4) [reply Formula/vis.rb]: No, that's a screw up on my part.

**Comment by DomT4** [comment]
Changes pushed.

File: Formula/vis.rb (hunk 2 above)
Comment (ilovezfs) [reply Formula/vis.rb]: Are we sure there's no `--progname=` for this? It seems really weird for upstream to use a system duplicate name and not make it configurable.

**Comment by ilovezfs** [comment]
:rocket:

**Comment by DomT4** [comment]
Thanks Joe. Here's hoping this is the last time I have to touch this formula for at least this week 😓.

**Comment by ilovezfs** [comment]
@DomT4 by the way, fx was asking why we don't compile `openssl` with `no-ssl3 no-comp`. any idea?

**Comment by DomT4** [comment]
The short answer is that it breaks API enough that you'd want to revision everything that `depends_on "openssl"`.

**Comment by ilovezfs** [comment]
ew

**Comment by DomT4** [comment]
It's something I've always wanted to get done but every time I've thought about it I've thought about the resulting CI job and ended up rocking silently in a corner. It's one of the reasons I was quite keen on shunting as much as possible onto `openssl@1.1` ASAP, but that obviously ran into other issues.

A previous minor hiccup is that until 1.0.2 was released `no-ssl3` didn't actually remove all of SSLv3, hence the creation of `no-ssl3-method` to "complete" the `no-ssl3` argument. Fun fun fun.

**Comment by ilovezfs** [comment]
I'm not really convinced of the merits of forcing a non-default value so :man_shrugging:

**Comment by DomT4** [comment]
For some prior on this: When I turned off SSLv2 in Homebrew's OpenSSL years ago now everything got revision bumped then, which went smoothly enough. MacPorts did not follow suit and left it enabled... and then OpenSSL _finally_ turned it off by default and _so many_ of MacPorts packages broke and needed recompiling 🙃.

**Comment by DomT4** [comment]
I think having SSLv3 on is bloody awful in terms of secure defaults, and OpenSSL agree enough to have disabled it by default in the 1.1 branch, but yeah, at this point I think we're stuck with the `openssl` formula as-is until that branch goes away permanently.

**Comment by DomT4** [comment]
> until that branch goes away permanently.

As a final note here: That ticking clock is the end of 2019.

**Comment by ilovezfs** [comment]
Hello libressl

**Comment by DomT4** [comment]
They more or less pulverised the API too 😅. It says a lot macOS has ended up using a mixture of LibreSSL, OpenSSL, BoringSSL, SecureTransport & CommonCrypto.

**Comment by ilovezfs** [comment]
I mean in 2019 what stops us from moving to libressl instead of 1.1

**Comment by DomT4** [comment]
In terms of practicality both would require a major effort. Neither are a drop in replacement for 1.0.2, and revision bumps everywhere would be a necessity. At that point it becomes a question of which has the widest support amongst formulae. I also suspect by then TLS 1.3 will be finalised, released & in demand and currently LibreSSL have taken no steps towards implementing it, so whether or not that changes will likely be a factor.

For better or worse there is nothing that is API-compatible enough with 1.0.2 to just drop in without it being a bloodbath.

**Comment by ilovezfs** [comment]
wonder how long until a 1.0 fork arises

**Pull request closed without merging**